tests/PHPStan/Analyser/nsrt/composer-array-bug.php

```diff
@@ -19,22 +19,22 @@ public function doFoo(): void
 			foreach ($this->config['authors'] as $key => $author) {
```

Comment:
Since we're doing `foreach ($this->config['authors'] as $key => $author)` we should be able to reduce the type of 
```
$this->config['authors']
```
to `array|object`.

This would avoid having later
```
assertType("array|ArrayAccess|string", $this->config['authors']);
```
when it cannot be a string.

Reply:
I think even `non-empty-array|Traversable`.

looking into it right now

Reply:
just committed a first iteration. it doesn't work in all cases yet.

Reply:
ok, I think we are in a usable state now.